proofs/proof_type.mli

```diff
@@ -26,3 +26,8 @@ type rule = prim_rule
 type goal = Goal.goal
 
 type tactic = goal sigma -> goal list sigma
+(** @deprecated: Please use "Tacmach.tactic" type instead.
+
+    We keep this definition only for "technical reasons".
+    If one tries to get rid of it, Ocaml compiler complains
+    about "circular dependency" between "Refiner" and "Tacmach" modules. *)
```

Comment:
Technically, the `tactic` type should eventually disappear, so it's not even recommended to switch to the `Tacmach` alias. Furthermore, the `Proof_type` module is probably one of the first modules that are going to disappear when we cut the old engine out of Coq.